sessionctx/variable/varsutil.go

```diff
@@ -162,6 +162,37 @@ func ValidateGetSystemVar(name string, isGlobal bool) error {
 	return nil
 }
 
+func checkIntegerSystemVar(name, value string, min int64, max uint64, vars *SessionVars) (string, error) {
```

Comment:
can we split this function to:
- `checkUInt64SystemVar`: check whether a `uint64` value is in the range [min, max]
- `checkInt64SystemVar`: check whether a `int64` value is in the range [min, max]

Reply:
Most checks here range from negatives to positive integers bigger than MaxInt64. We must take both negatives and big positives together, so we can't split this function into two....

Reply:
Besides, in MySQL, system variables that with positive ranges accept negative integer, truncate them to minimum legal value and raise a warning. so we should deal with singed and unsigned together.

Reply:
Is there a system variable ranged from negative integer to positive integer larger than `MaxInt64`?

If not, I think we can use `checkInt64SystemVar` and `checkUInt64SystemVar` to check the input parameters according to the valid value domain of the system variable. Handle the overflow error directly in these functions, which helps us to make code logic more clearer.

Reply:
ok